aiopg/pool.py

```diff
@@ -66,9 +68,11 @@ def __init__(self, dsn, minsize, maxsize, loop, timeout, *,
         self._enable_uuid = enable_uuid
         self._echo = echo
         self._on_connect = on_connect
+        self._pool_recycle = pool_recycle
         self._conn_kwargs = kwargs
         self._acquiring = 0
         self._free = collections.deque(maxlen=maxsize or None)
+        self._free_timestamps = {}
```

Comment:
I was thinking about that, but `Connection` doesn't know whether it's free or not at the moment, that I can tell. You'd be happy with an additional attribute which only gets updated through the pool?

I was going to add a `free` method to Connection to handle updating the time